tests/test_bgp_speaker.py

```diff
@@ -90,72 +88,104 @@ def test_bgp(localhost, ansible_adhoc, testbed, ipv4, ipv6, mtu):
+
+    ptfhost.script("./scripts/remove_ip.sh")
+
+    logging.info("########### Done teardown for bgp speaker testing ###########")
+
 
+def test_bgp_speaker_bgp_sessions(common_setup_teardown, duthost, ptfhost):
+    """Setup bgp speaker on T0 topology and verify bgp sessions are established
+    """
+    ptfip, mg_facts, interface_facts, vlan_ips, speaker_ips, port_num = common_setup_teardown
+
+    logging.info("Wait some time to verify that bgp sessions are established")
+    time.sleep(20)
+    bgp_facts = duthost.bgp_facts()['ansible_facts']
+    assert all([v["state"] == "established" for _, v in bgp_facts["bgp_neighbors"].items()]), \
+        "Not all bgp sessions are established"
+    assert str(speaker_ips[2].ip) in bgp_facts["bgp_neighbors"], "No bgp session with PTF"
+
+
+@pytest.mark.parametrize("ipv4, ipv6, mtu", [pytest.param(True, False, 1514)])
+def test_bgp_speaker_announce_routes(common_setup_teardown, duthost, ptfhost, ipv4, ipv6, mtu):
+    """Setup bgp speaker on T0 topology and verify routes advertised by bgp speaker is received by T0 TOR
+
+    """
+    ptfip, mg_facts, interface_facts, vlan_ips, speaker_ips, port_num = common_setup_teardown
+
+    logging.info("announce route")
     peer_range = mg_facts['minigraph_bgp_peers_with_range'][0]['ip_range'][0]
+    lo_addr = mg_facts['minigraph_lo_interfaces'][0]['addr']
+    lo_addr_prefixlen = int(mg_facts['minigraph_lo_interfaces'][0]['prefixlen'])
     prefix = '10.10.10.0/26'
-
-    # announce route
     announce_route(ptfip, lo_addr, prefix, vlan_ips[1].ip, port_num[0])
     announce_route(ptfip, lo_addr, prefix, vlan_ips[2].ip, port_num[1])
     announce_route(ptfip, lo_addr, peer_range, vlan_ips[0].ip, port_num[2])
 
-    # make sure routes announced to dynamic bgp neighbors
-    time.sleep(60)
-
-    bgp_facts = host.bgp_facts()['ansible_facts']
+    logging.info("Wait some time to make sure routes announced to dynamic bgp neighbors")
+    time.sleep(30)
 
-    # Verify bgp sessions are established
-    for k, v in bgp_facts['bgp_neighbors'].items():
-        assert v['state'] == 'established'
+    # The ping here is workaround for known issue:
+    #     https://github.com/Azure/SONiC/issues/387 Pre-ARP support for static route config
+    # When there is no arp entry for next hop, routes learnt from exabgp will not be set down to ASIC
+    # Traffic to prefix 10.10.10.0 will be routed to vEOS VMs via default gateway.
+    duthost.shell("ping %s -c 3" % vlan_ips[1].ip)
+    duthost.shell("ping %s -c 3" % vlan_ips[2].ip)
+    time.sleep(5)
 
-    # Verify accepted prefixes of the dynamic neighbors are correct
+    logging.info("Verify accepted prefixes of the dynamic neighbors are correct")
+    bgp_facts = duthost.bgp_facts()['ansible_facts']
     for ip in speaker_ips:
         assert bgp_facts['bgp_neighbors'][str(ip.ip)]['accepted prefixes'] == 1
     assert bgp_facts['bgp_neighbors'][str(vlan_ips[0].ip)]['accepted prefixes'] == 1
```

Comment:
This is redundant as vlan_ips[0] is in speaker_ips.
Could you, please, remove this to avoid confusion.

https://github.com/Azure/sonic-mgmt/pull/1326/files#diff-b204c198f5c49cbc64a65e74e5e9ea9eR61

Reply:
Right, I'll remove it. Thanks!